src/main/io/ledstrip.c

```diff
@@ -596,6 +598,54 @@ static void applyLedWarningLayer(bool updateNow, uint32_t *timer)
     }
 }
 
+static void applyLedBatteryLayer(bool updateNow, uint32_t *timer)
+{
+    static bool batteryFlash = false;
+
+    batteryState_e battState = getBatteryState();
+    if (updateNow) {
+        // keep counter running, so it stays in sync with blink
+        if (battState == BATTERY_OK) {
+            batteryFlash = true;
+            *timer += LED_STRIP_HZ(1);
+        }
+        else {
+            batteryFlash = !batteryFlash;
+            *timer += LED_STRIP_HZ(1.0 + calculateBatteryWarningPercentage() / 10.0);
+        }
+    }
+
+    const hsvColor_t *batteryColor =  &HSV(BLACK);
+
+    if (batteryFlash) {
+        switch (battState) {
+            case BATTERY_OK:
+                batteryColor = getSC(LED_SCOLOR_BATTERYOK);
+                break;
+            case BATTERY_WARNING:
+                batteryColor = getSC(LED_SCOLOR_BATTERYWARNING);
+                break;
+            case BATTERY_CRITICAL:
+                batteryColor = getSC(LED_SCOLOR_BATTERYCRITICAL);
+                break;
+            default:
```

Comment:
default: should not happen, so it makes no sense to set color here (HSV(BLACK) is fine)